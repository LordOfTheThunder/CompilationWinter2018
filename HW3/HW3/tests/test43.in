struct potato {
	int a;
	bool b;
	byte c;
	int d;
};
int PutToPotato(int a, bool b, byte c, int d, struct potato pot) {
	pot.a = a;
	pot.b = b;
	pot.c = c;
	pot.d = d;
}

int ReturnIntFunc() {
	int a = 3;
}

void callPutPotato() {
	struct potato pot;
	PutToPotato(ReturnIntFunc(), false, 33b, 22b, pot);
}


void main() { print("Max is cool"); }
