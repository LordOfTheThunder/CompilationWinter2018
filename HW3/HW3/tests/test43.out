---end scope---
a INT -1
b BOOL -2
c BYTE -3
d INT -4
pot struct potato -8
---end scope---
a INT 0
---end scope---
pot struct potato 0
---end scope---
---end scope---
print (STRING)->VOID 0
printi (INT)->VOID 0
struct potato {INT a,BOOL b,BYTE c,INT d}
PutToPotato (INT,BOOL,BYTE,INT,potato)->INT 0
ReturnIntFunc ()->INT 0
callPutPotato ()->VOID 0
main ()->VOID 0
